src/interpreter/virtualMachine.ts

```diff
@@ -0,0 +1,144 @@
+  }
+  return state.stash.size() === 0 ? undefined : state.stash.peek();
+};
+
+const virtualMachineEvaluators: VirtualMachineMapping = {
+  Assign: (instr: AssignInstr, state: VirtualMachineState) => {
+    // TODO: Add conversion method to convert various stash values to their respective number.
+    // Do this when types are supported.
+    state.memory.setByOffset(instr.scope, instr.offset, state.stash.pop());
+    state.memory.moveToNextInstr();
+  },
+  BinaryOperation: (
+    instr: BinaryOperationInstr,
+    state: VirtualMachineState
+  ) => {
+    const right = state.stash.pop();
+    const left = state.stash.pop();
+    typeCheckBinaryOperation(instr.operator, left, right);
+    state.stash.push(evaluateBinaryExpression(instr.operator, left, right));
+    state.memory.moveToNextInstr();
+  },
+  Call: (instr: CallInstr, state: VirtualMachineState) => {
+    // First item popped from the stash should be the arg for the first param and so on.
+    const args: Value[] = [];
+    for (let i = 0; i < instr.numOfArgs; i++) {
+      args.push(state.stash.pop());
+    }
+    const functionInstrAddress = convertToAddress(state.stash.pop());
+    state.memory.stackFunctionCallAllocate(args, instr.numOfVars);
+    state.memory.moveToInstr(functionInstrAddress);
+  },
+  Done: (instr: DoneInstr, state: VirtualMachineState) => {},
+  EnterProgram: (instr: EnterProgramInstr, state: VirtualMachineState) => {
+    state.memory.dataAllocate(instr.numOfDeclarations);
+    state.memory.moveToNextInstr();
+  },
+  Goto: (instr: GotoInstr, state: VirtualMachineState) => {
+    state.memory.moveToInstr(instr.instrAddress);
+  },
+  JumpOnFalse: (instr: JumpOnFalseInstr, state: VirtualMachineState) => {
+    const predicate = convertToPredicate(state.stash.pop());
+    if (isTrue(predicate)) {
+      state.memory.moveToNextInstr();
+    } else {
+      state.memory.moveToInstr(instr.instrAddress);
+    }
+  },
+  LoadConstant: (instr: LoadConstantInstr, state: VirtualMachineState) => {
+    state.stash.push(instr.value);
+    state.memory.moveToNextInstr();
+  },
+  LoadFunction: (instr: LoadFunctionInstr, state: VirtualMachineState) => {
+    state.stash.push(instr.functionInstrAddress);
+    state.memory.moveToNextInstr();
+  },
+  LoadSymbol: (instr: LoadSymbolInstr, state: VirtualMachineState) => {
+    const value = state.memory.getByOffset(instr.scope, instr.offset);
+    state.stash.push(value);
+    state.memory.moveToNextInstr();
+  },
+  Teardown: (instr: TeardownInstr, state: VirtualMachineState) => {
+    // Topmost return value should be from the rightmost return argument. We only want that.
+    if (instr.numOfReturnArgs > 0) {
+      const returnValue = state.stash.pop();
+      let numOfUnusedReturnValues = instr.numOfReturnArgs - 1;
+      while (numOfUnusedReturnValues > 0) {
+        state.stash.pop();
+        numOfUnusedReturnValues -= 1;
+      }
+      state.stash.push(returnValue);
+    }
```

Comment:
Instead of doing this, I feel like the multiple return values should be handled as part of `SequenceExpression` instead because we will see this pattern in multiple places. But I can do it in a separate PR.

Reply:
Okay I shall just write a TODO comment